x-pack/plugins/ml/public/explorer/explorer_utils.js

```diff
@@ -0,0 +1,490 @@
+  const jobIds = (selectedCells !== null && selectedCells.fieldName === VIEW_BY_JOB_LABEL) ?
+    selectedCells.lanes : getSelectedJobIds(jobs);
+  const timeRange = getSelectionTimeRange(selectedCells, interval);
+
+  if (mlAnnotationsEnabled === false) {
+    return Promise.resolve([]);
+  }
+
+  const resp = await ml.annotations.getAnnotations({
+    jobIds,
+    earliestMs: timeRange.earliestMs,
+    latestMs: timeRange.latestMs,
+    maxAnnotations: ANNOTATIONS_TABLE_DEFAULT_QUERY_SIZE
+  });
+
+  const annotationsData = [];
+  jobIds.forEach((jobId) => {
+    const jobAnnotations = resp.annotations[jobId];
+    if (jobAnnotations !== undefined) {
+      annotationsData.push(...jobAnnotations);
+    }
+  });
+
+  return Promise.resolve(
+    annotationsData
+      .sort((a, b) => {
+        return a.timestamp - b.timestamp;
+      })
+      .map((d, i) => {
+        d.key = String.fromCharCode(65 + i);
+        return d;
+      })
+  );
+}
+
+export async function loadAnomaliesTableData(selectedCells, jobs, dateFormatTz, interval, fieldName) {
+  const jobIds = (selectedCells !== null && selectedCells.fieldName === VIEW_BY_JOB_LABEL) ?
+    selectedCells.lanes : getSelectedJobIds(jobs);
+  const influencers = getSelectionInfluencers(selectedCells, fieldName);
+  const timeRange = getSelectionTimeRange(selectedCells, interval);
+
+  return new Promise((resolve, reject) => {
+    ml.results.getAnomaliesTableData(
+      jobIds,
+      [],
+      influencers,
+      mlSelectIntervalService.state.get('interval').val,
+      mlSelectSeverityService.state.get('threshold').val,
+      timeRange.earliestMs,
+      timeRange.latestMs,
+      dateFormatTz,
+      ANOMALIES_TABLE_DEFAULT_QUERY_SIZE,
+      MAX_CATEGORY_EXAMPLES
+    ).then((resp) => {
+      const anomalies = resp.anomalies;
+      const detectorsByJob = mlJobService.detectorsByJob;
+      anomalies.forEach((anomaly) => {
+        // Add a detector property to each anomaly.
+        // Default to functionDescription if no description available.
+        // TODO - when job_service is moved server_side, move this to server endpoint.
+        const jobId = anomaly.jobId;
+        const detector = get(detectorsByJob, [jobId, anomaly.detectorIndex]);
+        anomaly.detector = get(detector,
+          ['detector_description'],
+          anomaly.source.function_description);
+
+        // For detectors with rules, add a property with the rule count.
+        if (detector !== undefined && detector.custom_rules !== undefined) {
+          anomaly.rulesLength = detector.custom_rules.length;
+        }
+
+        // Add properties used for building the links menu.
+        // TODO - when job_service is moved server_side, move this to server endpoint.
+        anomaly.isTimeSeriesViewDetector = isTimeSeriesViewDetector(mlJobService.getJob(jobId), anomaly.detectorIndex);
+        if (has(mlJobService.customUrlsByJob, jobId)) {
```

Comment:
to remove another lodash include :)
this could be `mlJobService.customUrlsByJob[jobId] !== undefined`

Reply:
Done in 7eb77f19ab.